Add CMake preset for LLM

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #11052

Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

larryliu0820 committed:
Add CMake preset for LLM

Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]

jathu commented on .ci/scripts/test_llama.sh:
```diff
@@ -152,21 +152,9 @@ which "${PYTHON_EXECUTABLE}"
-        -DEXECUTORCH_BUILD_XNNPACK="$XNNPACK" \
-        -DEXECUTORCH_BUILD_MPS="$MPS" \
-        -DEXECUTORCH_BUILD_COREML="$COREML" \
+    retry cmake --preset llm \
```
it's beautiful 🥹

jathu commented on .ci/scripts/test_llama.sh:
```diff
@@ -152,21 +152,9 @@ which "${PYTHON_EXECUTABLE}"
-        -DEXECUTORCH_BUILD_MPS="$MPS" \
-        -DEXECUTORCH_BUILD_COREML="$COREML" \
+    retry cmake --preset llm \
         -DEXECUTORCH_BUILD_QNN="$QNN" \
```
I think you'll need to put these defines *before* the preset command

larryliu0820 committed:
Update on "Add CMake preset for LLM"


Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]

larryliu0820 committed:
Update on "Add CMake preset for LLM"


Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]

larryliu0820 committed:
Update on "Add CMake preset for LLM"


Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]

jathu reviewed (approved):
thank you!!!

jathu commented on .ci/scripts/test_llama.sh:
```diff
@@ -152,21 +152,11 @@ which "${PYTHON_EXECUTABLE}"
     rm -rf cmake-out
-    retry cmake \
+    retry cmake --preset llm \
         -DCMAKE_INSTALL_PREFIX=cmake-out \
```
This can be deleted too, since it's the default

larryliu0820 committed:
Update on "Add CMake preset for LLM"


Summary: By default turn on custom ops, optimized kernels, xnnpack. For
macOS/iOS turn on coreml and MPS.

Test Plan: Rely on CI jobs.

[ghstack-poisoned]
